File: csharp/PropertySearchResponder.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class PropertySearchResponder : ResultResponder
	{
		public override string Title()
		{
			return "Property Search: " + SearchForTags().ToString();
		}

		public override bool Traverse(WikiPage page)
		{
			List<string> commonTags = new List<string>();
			foreach (string tag in SearchForTags())
				if (page.Tags().Contains(tag))
					commonTags.Add(tag);
			return commonTags.Count > 0;
		}

		private string[] SearchForTags() {
			return request.Data["tags"].Split('/');
		}
	}
}

File: java/src/PropertySearchResponder.java
import java.util.ArrayList;
import java.util.List;


class PropertySearchResponder extends ResultResponder {
	public String title() {
		return "Property Search: " + search_for_tags().toString();
	}

	public boolean traverse(WikiPage page) {
		List<String> commonTags = new ArrayList<String>();
		for (String tag : search_for_tags()) {
			if (page.tags().contains(tag))
				commonTags.add(tag);
		}
		return commonTags.size() > 0;
	}

	private String[] search_for_tags() {
		return request.data().get("tags").split("/");
	}
}